etk-asm/src/asm.rs

```diff
@@ -353,58 +353,43 @@ impl Assembler {
     }
 
     /// Feed a single instruction into the `Assembler`.
-    ///
-    /// Returns the number of bytes that can be collected with [`Assembler::take`]
-    pub fn push<O>(&mut self, rop: O) -> Result<usize, Error>
+    fn push<O>(&mut self, rop: &O) -> Result<usize, Error>
     where
         O: Into<RawOp>,
     {
         let rop = rop.into();
 
-        self.declare_content(&rop)?;
-
-        // Expand instruction macros immediately. We do this here because it's the same process
-        // regardless if we `push_read` or `push_pending` -- in fact, `expand_macro` pushes each op
-        // individually which calls the correct unchecked push.
-        if let RawOp::Op(AbstractOp::Macro(ref m)) = rop {
-            self.expand_macro(&m.name, &m.parameters)?;
-            return Ok(self.ready.len());
-        }
-
-        self.push_unchecked(rop)?;
-        Ok(self.ready.len())
-    }
-
-    fn push_unchecked(&mut self, rop: RawOp) -> Result<(), Error> {
-        if self.pending.is_empty() && self.pending_len.is_some() {
-            self.push_ready(rop)
-        } else {
-            self.push_pending(rop)
-        }
-    }
+        self.declare_label(&rop)?;
 
-    fn push_ready(&mut self, rop: RawOp) -> Result<(), Error> {
         match rop {
             RawOp::Op(AbstractOp::Label(label)) => {
+                let mut dst = 0;
+                for ul in self.undeclared_labels.iter() {
+                    if (ul.label == label) & ul.dynamic_push {
+                        // Compensation in case label was dynamically pushed.
+                        let mut tmp =
+                            ((self.concrete_len as f32 - ul.position as f32) / 256.0).floor();
```

Comment:
Why use floating point for this? It seems equivalent to `(self.concrete_len - ul.position) / 256`. Integer division in Rust always floors.